test/language/expressions/async-generators/throw-suspendedYield.js

```diff
@@ -0,0 +1,39 @@
+// Copyright 2017 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Caitlin Potter <caitp@igalia.com>
+esid: pending
+description: >
+  AsyncGeneratorResumeNext:
+  If completion.[[Type]] is throw, and generator.[[AsyncGeneratorState]] is
+  "suspendedYield", generator is resumed and immediately and
+  closes the generator and returns completion.
+flags: [async]
+---*/
+
+class Err extends Error {}
+var g = async function*() {
+  yield 1;
+  $ERROR('Generator must not be resumed.');
+};
+
+var it = g();
+it.next().then(function(ret) {
+  // Initial yield
+  assert.sameValue(ret.value, 1);
+  assert.sameValue(ret.done, false);
```

Comment:
I've done this --- but it gets complicated for the generator-closing operations. I've gone with using the spec step which leads to the promise capability resolve/reject call instead there, but that seems awkward too.